### ISSUE TITLE
macOS 11.0 pools will be transited to private preview.

### ISSUE THREAD

**Issue by AlenaSviridenko** [actions/virtual-environments#2486]
Hi GitHub Actions community!

A couple of months ago we added a new macOS 11.0 image in a “preview” mode and eventually we expect it to become a production image.

Typically, we increase pool capacity along with increasing image usage, but demand for this pool is outpacing our capacity plan.
Given the constantly growing demand and in order to prevent degraded experience, we have to temporarily transit this pool to a private preview until we come up with a final plan of making macOS 11.0 ready for production use.

_What does “private preview” mean?_
This limitation will live on organization level (personal accounts are considered as organization too), so if you are using `macos-11` somewhere in your organization — you will still be able to use it. But this pool will be inaccessible for new organizations that never used it before.

We are sorry for inconveniences, and will try to open this pool again as soon as possible.

### UPDATE
The macOS 11 virtual environment is currently in preview and is automatically available to the existing Enterprise plan customers who used macOS hosted runners at least once between May, 1 - June, 1. New Enterprise plan customers, or customers on other plans, should fill the form to request access to macOS 11 virtual environment. Please view our [Big Sur guide](https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md) for more details.

**Comment by tomhamming** [comment]
The docs still mention `macos-11.0`: 

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

**Comment by AlenaSviridenko** [comment]
Hi @tomhamming,
thanks for noticing it, we will update docs soon as well.

**Comment by rwwagner90** [comment]
Would this have anything to do with my macos builds always hanging now?

**Comment by AlenaSviridenko** [comment]
Hi @rwwagner90,
closing pool was the first step in resolving constantly increasing queues, and now we are looking into how to improve current queues given the capacity we have. Hopefully, queues will become more healthy soon.
Thank you for your patience.

**Comment by rwwagner90** [comment]
@AlenaSviridenko thanks, hope things improve soon! I wasted a lot of hours yesterday trying to debug this, but I think it was just the Mac images not working.

**Comment by paulz** [comment]
macOS 11 image works great for us. We are using GitHub actions as our CI to build and release macOS 11 app. 


When the preview is planned to be a production?

**Comment by AlenaSviridenko** [comment]
Hi! Currently, we don't have any certain ETA, but we will announce officially and publicly when macOS 11.0 will be ready for production.

**Comment by christian-kreuzberger-dtx** [comment]
Hi,
big thanks for the ongoing support for macos 11.0, we're using it in keptn project for building our CLI.

coming from travis-ci, I always liked this status page https://www.traviscistatus.com/, most importantly the "backlog" (= queue) of macos builds:
![image](https://user-images.githubusercontent.com/56065213/106619174-94ac6280-6570-11eb-98dc-aca4b12ecc1e.png)

Is there such a thing for virtual environments on GH Actions? This would really help users to troubleshoot problems (e.g., oh my pipeline is stuck because MacOS builds are queued)

**Comment by paulz** [comment]
Thank you for supporting latest macOS release. As a customer we would like GitHub to make public releases of runners when Apple makes macOS release public. For example macOS 11 was released November 2020. Meantime we are happy to use beta or preview versions.

**Comment by KeiroMidori** [comment]
How does one know if they are being part of new or existing organizations using `macos-11.0` ? Will there be verbose errors saying that they should use `macos-latest` instead?

**Comment by radarhere** [comment]
> How does one know if they are being part of new or existing organizations using `macos-11.0` ? Will there be verbose errors saying that they should use `macos-latest` instead?

Over at https://github.com/python-pillow/pillow-wheels/runs/1813985728?check_suite_focus=true, I'm see 'X This check failed' - contrasted with the successful run at https://github.com/radarhere/pillow-wheels/runs/1813987594?check_suite_focus=true, where my fork is in the pool and runs without a problem.

**Comment by AlenaSviridenko** [comment]
> How does one know if they are being part of new or existing organizations using `macos-11.0` ? Will there be verbose errors saying that they should use `macos-latest` instead?

Hi! if organization used `macos-11.0` last 30 days — pool should be available. If it is not — `No runner matching the specified labels was found: macos-11.0.` error will be displayed.

**Comment by KeiroMidori** [comment]
> > How does one know if they are being part of new or existing organizations using `macos-11.0` ? Will there be verbose errors saying that they should use `macos-latest` instead?
> 
> Hi! if organization used `macos-11.0` last 30 days — pool should be available. If it is not — `No runner matching the specified labels was found: macos-11.0.` error will be displayed.

Roger that. Apparently my organization is able to use `macos-11.0` but it's really unstable. Queue times are enormous and workflows often fail. I went back to using `macos-latest` for now but I'd love to see these private previews actually working for those who can use them.

**Comment by AlenaSviridenko** [comment]
Hi @christian-kreuzberger-dtx,
unfortunately GH Actions doesn't provide such status page, but the idea is interesting, we'll take a look into it, thanks!

**Comment by erikng** [comment]
@AlenaSviridenko my organization had used the macos-11.0 image (11 times) with a latest use of 2 months ago. We maintain a custom python that is used for tens of thousands of enterprise macOS devices. We hadn't had to build python until today and now it will be impossible to maintain the project on GitHub.

I need access to the image because it's the only way to reliably build universal macOS python wheels that support both Intel and Apple Silicon macs.

Is there anyway to get this re-allowed? The project is located here: https://github.com/macadmins/python

**Comment by paulz** [comment]
we are waiting for a long time for runner to pick up jobs on `macos-11.0` to run our CI tests.

if it helps for anyone we found that `macos-latest` runner can at least compile and build MacOS 11.0 application. So we are using more `macos-latest` runner to build and publish releases for macOS 11 even it is unable to run tests or even load the app.

**Comment by csett86** [comment]
@erikng if it helps, this is how I built universal binaries with the current macos-latest (10.15) runner: https://github.com/csett86/jitsi-meet-electron/blob/main/.github/workflows/nodejs.yml#L58

**Comment by wez** [comment]
FWIW, I suspect that a lot of CI use cases just want to build with the newer SDK (mine fit in this category), so if the docs made it clearer how to do that, then I suspect that there would be less demand for 11.0 itself.

**Comment by dennisameling** [comment]
> @erikng if it helps, this is how I built universal binaries with the current macos-latest (10.15) runner: https://github.com/csett86/jitsi-meet-electron/blob/main/.github/workflows/nodejs.yml#L58

That works, thanks @csett86! 🚀

**Comment by erikng** [comment]
The problem is two fold for us. One is compiling the wheels and then the secondary is creating the package. From how I understand it, only product build on Big Sur can create a package with a distribution file that supports both Intel and Apple silicon Macs. If you build a package on Catalina, it will be universal (and can be notarized) but it will trigger a user to need to install Rosetta2 on AS Macs. 

Those tools are built into the OS and not part of the Xcode command line tools package.

**Comment by tonyarnold** [comment]
GitHubbers, could you please provide an update on this issue? It's been 3 months since Big Sur's public release - I understand that there may be extenuating circumstances here, but if this is the speed with which the Actions team is going to be deploying major macOS updates, then Hosted Runners aren't going to be a viable option for those of us trying to use your service to build macOS apps. 

As I'm sure you are also aware, Xcode 12.5 is now in beta, and it _requires_ macOS 11. Things are about to become untenable for many iOS developers using GitHub Actions who want to test on the latest SDKs.

**Comment by AlenaSviridenko** [comment]
Hello,

we apologize for a long silence, we understand that macOS 11.0 is a highly demanded OS now because of the latest Xcode, unique Swift UI features and so on, this is our 1st priority for now, and although it appeared to be a pretty tricky from the all point of view, we are almost on the final line with the support and capacity plan. We will come up with this plan by the end of February.

Thank you for your patience.

**Comment by Nenginyo0804** [comment]
> Hi GitHub Actions community!
> 
> A couple of months ago we added a new macOS 11.0 image in a “preview” mode and eventually we expect it to become a production image.
> 
> Typically, we increase pool capacity along with increasing image usage, but demand for this pool is outpacing our capacity plan.
> Given the constantly growing demand and in order to prevent degraded experience, we have to temporarily transit this pool to a private preview until we come up with a final plan of making macOS 11.0 ready for production use.
> 
> _What does “private preview” mean?_
> This limitation will live on organization level (personal accounts are considered as organization too), so if you are using `macos-11.0` somewhere in your organization — you will still be able to use it. But this pool will be inaccessible for new organizations that never used it before.
> 
> We are sorry for inconveniences, and will try to open this pool again as soon as possible.

**Comment by AlenaSviridenko** [comment]
Hi there,
sad to say, but not too many updates. Unfortunately, we still do not have the ability to take new users onto the preview pool. We expect the earliest we can is in April, when we're first able to start adding new capacity.

**Comment by QuiiBz** [comment]
We will wait - but is it normal to still have `macos-11.0` listed in the [Supported runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)? I think it should advertise that no new users can be added to the preview pool.

**Comment by erikng** [comment]
@AlenaSviridenko is there any chance you could help my org? We host packages for enterprise customers around the world. As I stated before we had access to the runners and per notes from GitHub all customers that had run them prior would not be impacted but when the change was deployed, that was not the case.

**Comment by AlenaSviridenko** [comment]
Hi @erikng, we included your org back to macOS-11.0 pool, sorry for inconveniences and late reply. We have some internal limitation of pool usage history, that is why your org was not included previously.

**Comment by erikng** [comment]
Thank you!

**Comment by EvgeneOskin** [comment]
Hi @AlenaSviridenko, we've run GitHub Action with macOS 11 runner before it moved to private preview. We used it to distribute the Termius app for the arm64 architecture.
Can we get access back to the macOS-11.0 pool? Could you help with this?

**Comment by kalcutter** [comment]
Please give us back access to macOS 11. Our organisation was using it successfully last year.

**Comment by sarahlovemike** [comment]
I need help don't know what I'm doing

**Comment by kode54** [comment]
Guess I'll just continue doing builds on my personal machine, and set my CI to only test Intel builds.

**Comment by freak4pc** [comment]
Sorry for jumping late to the party but this is quite silly. At the bare minimum you should remove this information from [virtual-environments](https://github.com/actions/virtual-environments) to avoid confusion and time waste :)

**Comment by AlexanderAmelkin** [comment]
I second freak4pc's suggestion. I must note also that if I add macos-11.0 and want to use a matrix of [ maxos-10.15, macos-11.0 ], then the 11.0 build never finishes without saying that it is not available, and the 10.15 build silently gets 'cancelled' without any explanation at all. That doesn't look like correct behavior. I think you must announce removal of 11.0 environment for new users, and remove 11.0 from the documentation until it the capacity is available. The current behavior is highly confusing.

**Comment by MarcoEidinger** [comment]
@AlenaSviridenko Would you be so kind to give a status update on the macOS 11.0 image? Today's Xcode 12.5 RC release (on macOS 11 only) makes the availability even more critical as the official release is expected for next week.

**Comment by AlenaSviridenko** [comment]
Hi 👋,
we're working on a plan to get Big Sur added to the hosted pool sometime in June/July, we are still working on hardware infrastructure to be ready. We are aware about stable Xcode 12.5 coming soon and doing everything possible to make macOS 11 ready as soon as possible.

**Comment by christian-kreuzberger-dtx** [comment]
Seems like MacOS 11 builds are currently not working even in private preview. They are failing with
```
An error occurred while provisioning resources (Error Type: Failure).
```

**Comment by sisoje** [comment]
> Hi 👋,
> we're working on a plan to get Big Sur added to the hosted pool sometime in June/July, we are still working on hardware infrastructure to be ready. We are aware about stable Xcode 12.5 coming soon and doing everything possible to make macOS 11 ready as soon as possible.

that is way too late

**Comment by freak4pc** [comment]
*Three months* for a new version of Xcode? That would be a month after WWDC. 
If you're trying to be an acceptable commercial CI solution, this kind of SLA will never work for developers.

Bitrise [https://discuss.bitrise.io/t/xcode-12-5-release-candidate-12e262-is-now-available-on-big-sur/16612](already has Xcode 12.5 RC1).

**Comment by tcldr** [comment]
> we're working on a plan to get Big Sur added to the hosted pool sometime in June/July

This is very frustrating for those of us who have invested significant time into GitHub Actions.

Should we expect a similar lead time for macOS 12 when it is inevitably announced in June/July?

**Comment by danpalmer** [comment]
There was a big [licencing change (#2604)](https://github.com/actions/virtual-environments/issues/2604) between macOS 10.15 and 11, specifically around how machines can be rented out. I'd assume that GitHub/Microsoft are probably building out infrastructure based on the new licencing (I'd expect to see Azure macOS VMs), and that will back Azure Pipelines and GitHub Actions once it's live. If this is the case this delay is likely a one-off issue rather than a pattern for the future.

I suspect this is not something we need to worry about, although it would certainly be nice to see it happen as soon as possible.

Anyone at GitHub/Microsoft please do correct me, if you're allowed to!

**Comment by AlenaSviridenko** [comment]
> Should we expect a similar lead time for macOS 12 when it is inevitably announced in June/July?

Hopefully, not. Initially, GitHub Actions started with a single macOS version, so adding a new one brought some challenges. But once we polish the onboarding plan, we could stick to it for future versions, so onboarding cadence should be way better.

**Comment by lrun-cmck** [comment]
Without macOS 11.0, we won't be able to use Xcode 12.5 which requires Big Sur 11 or later (https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes)

**Comment by scannillo** [comment]
> Without macOS 11.0, we won't be able to use Xcode 12.5 which requires Big Sur 11 or later (https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes)

Also worth adding we need Big Sur in order to test iOS 14.5 and up.

**Comment by ffried** [comment]
> > Without macOS 11.0, we won't be able to use Xcode 12.5 which requires Big Sur 11 or later (https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes)
> 
> Also worth adding we need Big Sur in order to test iOS 14.5 and up.

Yep. Also, it makes it a lot harder to use Swift 5.4 on macOS.
I've hit this today while updating our OSS components - which are now all untested on the newest Xcode / Swift 5.4.

**Comment by rmaloney3** [comment]
I agree with all of the comments said thus far on the topic. macOS 11 needs to be opened up much sooner than June/July. We need the ability to use Xcode 12.5 on CI, as we have plans imminent plans to start using Swift 5.4 features. 

I should also add that Apple has already released a beta version of iOS 14.6 which also relies on Xcode 12.5. We also all know that at WWDC on June 7, Apple will be releasing betas for Xcode 13, macOS 12, iOS 15. These need to be available on CI before July as well. 

I have to say I expected more from GitHub and Microsoft.

**Comment by nzjrs** [comment]
What the hell? The docs still mention this as available and yet I have to dig around here to find out that the semi-silent failure of my matrix build on macos-11 was actually by design.

**Comment by michaelpeternell** [comment]
Sorry, maybe I am saying something obvious now..

Do you know that you don't need Apple Silicon to run macOS 11?

I would of course appreciate a virtual environment with macOS 11.3 running on Apple Silicon. But a virtual environment with macOS 11.3, running on an Intel Mac, with Xcode 12.5, would be fine for now, at least for me. And it would certainly be better than no macOS 11 support at all.

**Comment by kneekey23** [comment]
agree with all of the above comments, need this for Xcode 12.5 in our CI now as we have to be on Swift 5.4 for real issues that were fixed in 5.3 specifically for our organization by Apple. If we can't be on Swift 5.4, we can't even work so need 12.5 in CI. How can we join the private preview as an org? I added the `macos-11.0` and also tried `macos-11.2` but I get an error in the check `This check failed` before it even starts building I presume because my org isn't apart of the preview? what do I need to do in order to join? Thanks!

**Comment by praveenperera** [comment]
Waiting on MacOs 11 to create binaries for M1 macs

**Comment by paulz** [comment]
> Waiting on MacOs 11 to create binaries for M1 macs

We've been using `xcodebuild` on macOS 10 to create binaries for M1. And our app is macOS 11 only. We only use macOS 11 for running automated tests.

**Comment by jarrodrobins** [comment]
@AlenaSviridenko I understand you and the GitHub Actions team very likely have a lot on your plate and I realise it's probably unhelpful to keep badgering you and your team about it. However, I feel it's important for us to emphasise to your _leadership_ that for the macOS and iOS development community, adding macOS 11 support is not a "nice to have". Not having it is a *blocker*. Waiting until June or July is way too long. Waiting until _May_ was too long.

Apple drops support for the previous operating system consistently each year. GitHub could and should have seen this coming.

We need to be able to use the latest version of Xcode to keep up to date with the latest bug fixes, features, and devices that Apple releases. 

Without it, the only choice is to move to a provider that treats iOS and macOS developers as a priority. When other providers such as Bitrise are providing Xcode _betas_ within a day or two, it makes me wonder what the benefit of continuing to use GitHub actions is when we can't even use release builds.

I would ask that as a minimum to please provide a more concrete timeline as to when macOS 11 will be available for all GitHub Actions users so we have a degree of certainty around this.

**Comment by michaelpeternell** [comment]
Does macOS 11 support take so long because Github tries to provide macOS 11 virtual machines on Apple Silicon macs? If they just install macOS 11 on an Intel Mac, there should be no problem, should it?

That's what I was trying to say with my [previous comment](https://github.com/actions/virtual-environments/issues/2486#issuecomment-831338313).

Can we have a statement by Github that either:
- yes they are trying to install macOS 11 on Apple Silicon, and that's the reason why it takes so long. And macOS 11 on Intel Macs would be easy.
- no, macOS 11 on Intel Macs is equally hard to do, and they've tried this already.

**Comment by oloastrickland** [comment]
The need for xCode 12.5 is pretty big. Apple requires the new Apple Tracking Transparency framework if you do any data tracking. That framework is only available in xCode 12.5. This means we cannot generate any builds (CI, testing or app store) using Azure Pipelines or Github actions. I think any real business or dev teams, including ours, will be looking at alternatives now.

**Comment by SLambrakis** [comment]
@oloastrickland  I completely agree that I'd love for for Github Actions to provide MacOS11/Xcode 12.5. However, just in case it helps you get unblocked, I believe that the ATT framework was actually available from iOS14.0/Xcode 12.0. You're only *required* to use it from iOS14.5, but it was available earlier than that if you chose to opt-in. So you might be able to continue to make builds using Xcode 12.4.

**Comment by AlenaSviridenko** [comment]
Setting up macOS 11 on Intel Macs is hard too, but we are really close to getting new capacity. Hopefully, it will happen sooner in May, this is our 1st priority for now.

**Comment by svaldetero** [comment]
Someone mistakenly installed Xamarin.iOS 14.16 on the macOS 10.15 image and set it to default. Xamarin.iOS 14.16 has a hard dependency on Xcode 12.5. From this thread, I found out that Xcode 12.5 can't be installed on macOS 10.15. All of my Xamarin.iOS builds are failing without making a change because of this discrepancy since everything is set to use the default / latest. 

I get this error when building
`MTOUCH : error MT0180: This version of Xamarin.iOS requires the iOS 14.5 SDK (shipped with Xcode 12.5). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs).`

Can we get this installation rolled back since I don't actually need macOS 10.11 and don't want to take from the limited resources. I also don't want to have to update all of my build definitions to pin to a previous version of Xamarin.iOS.

**Comment by pdblasi-billgo** [comment]
@svaldetero 

For an immediate fix, you can add this step to your iOS build job to put [this Action](https://github.com/marketplace/actions/setup-xamarin) into your pipeline, which will set the version of Xamarin.iOS used to a specific version.

```
    - name: Setup Xamarin
      uses: maxim-lobanov/setup-xamarin@v1
      with:
        xamarin-ios-version: '14.14'
        xcode-version: '12.4'
```

I agree that it should be rolled back anyways, since this isn't a useful Xamarin configuration for this OS, but that may get you up and running a bit quicker.

**Comment by AlenaSviridenko** [comment]
We have already merged the fix https://github.com/actions/virtual-environments/pull/3375, and it will be available with the next macOS deployment approximately on Wednesday.
We are investigating how that happened because our internal tests on macOS 10.15 with the configuration of Xamarin.iOS 14.16 + Xcode 12.4 have passed. Maybe some more complex cases cause this error

**Comment by svaldetero** [comment]
@AlenaSviridenko Thanks for the quick fix! I'm glad that is has already been identified and scheduled for deployment.

**Comment by richieshilton** [comment]
@AlenaSviridenko 👋 are there any updates for this or updates on a revised ETA? thanks

**Comment by michaelpeternell** [comment]
I don't really understand why Github can't just take an existing VM with macOS 10.15 and hit the "Update" button that is under Settings-app, then "Software Update". Can someone explain why this OS update is different? Am I missing something obvious here? Because that is how I installed macOS 11 on my local machine.

**Comment by caioremedio** [comment]
> I don't really understand why Github can't just take an existing VM with macOS 10.15 and hit the "Update" button that is under Settings-app, then "Software Update". Can someone explain why this OS update is different? Am I missing something obvious here? Because that is how I installed macOS 11 on my local machine.

If you take time to read this thread you may find out why.

> Setting up macOS 11 on Intel Macs is hard too, but we are really close to getting new capacity. Hopefully, it will happen sooner in May, this is our 1st priority for now.
https://github.com/actions/virtual-environments/issues/2486#issuecomment-832148492

**Comment by michaelpeternell** [comment]
@caioremedio : Your reply is just rude, I would prefer if you don't participate in this conversation.

> > Setting up macOS 11 on Intel Macs is hard too, but we are really close to getting new capacity. Hopefully, it will happen sooner in May, this is our 1st priority for now.
> > [#2486 (comment)](https://github.com/actions/virtual-environments/issues/2486#issuecomment-832148492)

I read everything. And this particular quote doesn't explain anything by the way.

**Comment by michaelpeternell** [comment]
Maybe Github can give us a little bit more information on what is going on? Where is the problem exactly?

How was the original pool organized? Did you just update individual macOS 10.15 machines manually? If so, why can't this be done with more machines? Or did you run macOS 11 on the new M1-macs? If so, can you order more of them? And did Apple deliver them? If yes, is there a problem with integrating them because the current data center is too small? I asked a similar question 2 weeks ago ( https://github.com/actions/virtual-environments/issues/2486#issuecomment-831762886 ) and that was just ignored.

Does anyone here know if the macOS 11 pool runs on Intel or on Apple Silicon? The answer quoted above ("Setting up macOS 11 on Intel Macs is hard too") seems to be carefully crafted to avoid answering that question.

**Comment by davdroman** [comment]
@michaelpeternell I don't think you'll get an answer about what exactly GitHub is doing operationally to not be able to deliver this on time. Whatever issues they're having due to M1, it's more than likely they're learning from this and they'll do their best not to have this happen again with the next OS release. Pretty much every org has struggled with the transition, both at hardware and software sides (e.g. Homebrew).

**Comment by allschu** [comment]
Working with Azure Devops, got the exact same problem

**Comment by caioremedio** [comment]
@michaelpeternell Sorry if I sound rude but your message doesn't has any tech question. Tbh you just sound arrogant to the entire Dev Team by saying: "Hey guys there is an Update Button inside System Prefs you know that? Why not pressing it?"

So yes, my quote explain exactly what you are looking for: Why aren't they pressing the update button?

And please avoid stating obvious information like if the dev team are all stupid.

**Comment by michaelpeternell** [comment]
@caioremedio : seems like everyone agrees that I am stupid. Okay, I got it. I'm still genuinely curious what the problem is. No one wants to tell me. I understand. I know that maybe they tried to press the "Update" button. And it probably failed with some error message. Or they update through some other means, maybe there is a command line interface for that. Maybe the answer to my question of wether the current macOS 11 pool runs on Intel or on Apple Silicon is obvious too, so you assume that I must be arrogant for asking that question. That would have been a tech question, by the way. Anyhow, I don't care anymore.

**Comment by ffried** [comment]
@michaelpeternell I think the process isn't as simple as "updating a machine".
I don't know how exactly GitHub maintains their runners, but I could imagine that they are transient virtual machines. Meaning (a bit simplified) every time a job defined to run on a macOS machine is submitted, they set up a new (clean) VM and tear it down once the job has finished.
To be able to do this, the process of setting up and tearing down a VM needs to be automated which I believe is the tricky part here. There isn't simply a machine that you can upgrade and use it from now on. A process needs to be set up that dynamically creates and destroys virtual machines as they are needed.

Nevertheless, I also hoped for a bit more transparency in this process so that we can understand why exactly it takes so long...

**Comment by allschu** [comment]
> @caioremedio : seems like everyone agrees that I am stupid. Okay, I got it. I'm still genuinely curious what the problem is. No one wants to tell me. I understand. I know that maybe they tried to press the "Update" button. And it probably failed with some error message. Or they update through some other means, maybe there is a command line interface for that. Maybe the answer to my question of wether the current macOS 11 pool runs on Intel or on Apple Silicon is obvious too, so you assume that I must be arrogant for asking that question. That would have been a tech question, by the way. Anyhow, I don't care anymore.

No, its not you...its a little vague for me also.
But my problem is the fact that Azure Devops is expecting image for my pipeline, that they have retracted I think.

**Comment by CamJN** [comment]
I'm guessing the difficulty stems from the azure folks needing to contend with the new EULA terms where you need to dedicate an entire host to a single user for 24hrs in Big Sur: https://blog.macstadium.com/blog/developers-big-sur-and-vindication

That doesn't play well with short lived CI vms at all.

**Comment by kode54** [comment]
It's probably because Apple probably wants all software development to be handled by developers on their personal machines, and put an end to this CI nonsense. But that's just my reading of it.

**Comment by rmaloney3** [comment]
> It's probably because Apple probably wants all software development to be handled by developers on their personal machines, and put an end to this CI nonsense. But that's just my reading of it.

You clearly didn't read @CamJN's article, as it states that this use case is now officially supported by Apple (albeit with some restrictions).

**Comment by jasonzurita** [comment]
> 👋 are there any updates for this or updates on a revised ETA? thanks

Is there a way to find out a revised ETA?

(I am sure the GH Actions team is doing everything they can to get this done as soon as possible, so thank you for all your effort on this!)

**Comment by AlenaSviridenko** [comment]
Hey,
we are targeting end of May / beginning of June to start opening pool for Enterprise and Teams customers, and later for all customers, but that is a pretty rough timeline and the actual dates can vary.

**Comment by RobinZergiebel** [comment]
For Azure DevOps users who end up here:

You need to add the following task to your yaml:

`
- script: sudo $AGENT_HOMEDIRECTORY/scripts/select-xamarin-sdk.sh 6_12_7
    displayName: 'Select symlink version'
    enabled: true
`

The default symlink version (6_12_8) uses Xamarin.iOS 14.16 which requires Xcode 12.5:

![image](https://user-images.githubusercontent.com/56680419/118779055-faf09300-b88a-11eb-973f-999ea1135bf5.png)

However, Xcode 12.5 will only be available with MacOS 11.0. Therefore you have to use symlink version 6_12_7.

Cheers,
Robin

**Comment by JetForMe** [comment]
We really need this support, too. We're willing to pay for access to runners.

**Comment by freak4pc** [comment]
> We really need this support, too. We're willing to pay for access to runners.

If that's the case, just get a MacStadium Mac Mini and run your own custom rubber (you can see ReactiveX/RxSwift for an example)

**Comment by MaxDesiatov** [comment]
That really isn't suitable for open-source projects since self-hosted runners aren't secure against external PRs. Even GitHub themselves [don't recommend using them in this case](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories):

>We recommend that you only use self-hosted runners with private repositories. This is because forks of your repository can potentially run dangerous code on your self-hosted runner machine by creating a pull request that executes the code in a workflow.

**Comment by shepting** [comment]
In case upgrading the machines is the bottleneck (which I suspect it is), I found that the following command can initiate the upgrade from the command-line: `sudo /Applications/Install\ macOS\ Big\ Sur.app/Contents/Resources/startosinstall --agreetolicense --forcequitapps --nointeraction`

You will still need to get that image placed on all of the Catalina machines, but that would be straightforward with Chef/Puppet/Terraform or whatever you've got running on there.

The bigger issue seems to be that even after running the upgrade, most of your services won't start up again until you log in through VNC (just an SSH terminal won't do it). I haven't yet found what other steps are required to fully automate this step.

**Comment by jshier** [comment]
> > If that's the case, just get a MacStadium Mac Mini and run your own self-hosted runner (you can see ReactiveX/RxSwift for an example)
> 
> That really isn't suitable for open-source projects since self-hosted runners aren't secure against external PRs. Even GitHub themselves [don't recommend using them in this case](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories):
> 
> > We recommend that you only use self-hosted runners with private repositories. This is because forks of your repository can potentially run dangerous code on your self-hosted runner machine by creating a pull request that executes the code in a workflow.

This doesn't really seem like an issue anymore, as GitHub now requires organizational approval to run CI on PRs from non-members. This is a great thing even without such abuse issues since it means we don't waste time running CI on documentation and typo fix PRs. But it should allow using hosted runners without issue.

**Comment by freak4pc** [comment]
> > If that's the case, just get a MacStadium Mac Mini and run your own self-hosted runner (you can see ReactiveX/RxSwift for an example)
> 
> 
> 
> That really isn't suitable for open-source projects since self-hosted runners aren't secure against external PRs. Even GitHub themselves [don't recommend using them in this case](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories):
> 
> 
> 
> >We recommend that you only use self-hosted runners with private repositories. This is because forks of your repository can potentially run dangerous code on your self-hosted runner machine by creating a pull request that executes the code in a workflow.

First of all, see @jshier's comment.

Second, as a temporary one month workaround it's overall fine in my book. I imagine very few use cases a consumer would open a PR to cause harm and a way that isn't very easy to revert given the machine only has the runner code and open source code publicly available.

**Comment by kode54** [comment]
Starting at $109/mo to rent a machine that costs just 7 times that? Not exactly what I'd call affordable for all but the dedicated and wealthy. I'll just continue churning out my own builds from home, in that case.

**Comment by freak4pc** [comment]
> Starting at $109/mo to rent a machine that costs just 7 times that? Not exactly what I'd call affordable for all but the dedicated and wealthy. I'll just continue churning out my own builds from home, in that case.

They also have open source machines. You need to apply for it, but still.

**Comment by kode54** [comment]
> > Starting at $109/mo to rent a machine that costs just 7 times that? Not exactly what I'd call affordable for all but the dedicated and wealthy. I'll just continue churning out my own builds from home, in that case.
> 
> They also have open source machines. You need to apply for it, but still.

Oh. Oops. I'd gave to see if that's useful for my future cases, except, that adds more complication where I don't exactly need it right now. But thanks for the info.

**Comment by freak4pc** [comment]
No problem. Ok just say it probably takes 10 minutes to set up.

**Comment by anfriis** [comment]
App Center apparently depends on this PR to be merged before they can support MacOS Big Sur and Xcode 12.5 and a lot of people are desperate for this. Hope you will find time to solve it soon 🥇  
[https://github.com/microsoft/appcenter/issues/2185](https://github.com/microsoft/appcenter/issues/2185)

**Comment by kneekey23** [comment]
can we please get another status update on this? I believe end of May was said at some point and I want to know if that timeline still stands. Thanks!

**Comment by lilliSeBe** [comment]
@AlenaSviridenko looking at https://github.com/actions/virtual-environments/issues/2486#issuecomment-843156334 I know @jonico has already contacted you about this. Where can I request for an Enterprise customer to be part of the first pool of users?

**Comment by richieshilton** [comment]
> we are targeting end of May / beginning of June...

Is there any update on this?

**Comment by MehdiChinoune** [comment]
For those asking for an update, see #3449

**Comment by gtalarico** [comment]
Our CI is blocked by this as well, please let me know if there is a way to access the preview!

**Comment by AlenaSviridenko** [comment]
Hi there!
Please, check out our project's [README](https://github.com/actions/virtual-environments#available-environments) for the latest updates about macOS 11.

**Comment by diogot** [comment]
Thanks for the update!
The documentation linked about Azure DevOps don't say anything about macOS 11, do this means that it's outdated or macOS 11 isn't available?

**Comment by AlenaSviridenko** [comment]
macOS 11 is currently not available for Public AzDO customers, but we are also working on providing it as soon as possible. Azure DevOps infrastructure a bit different so the amount of work required is also different.

**Comment by jpsim** [comment]
@AlenaSviridenko your [latest comment](https://github.com/actions/virtual-environments/issues/2486#issuecomment-852906007) made it seem like macOS 11 images were now publicly available for GitHub Actions even outside the private preview.

The readme you linked to also indicates macos-11 as being 100% rolled out.

![image](https://user-images.githubusercontent.com/474794/120493241-a27fc200-c36f-11eb-81d1-7e4a4eb8a48d.png)

However, GitHub Actions [still aren't running macos-11 jobs](https://github.com/jpsim/Yams/runs/2728372207?check_suite_focus=true):

> No runner matching the specified labels was found: macos-11

![image](https://user-images.githubusercontent.com/474794/120493425-ccd17f80-c36f-11eb-8df6-b21c182f0772.png)

Am I misinterpreting your comment and the readme? Could you please clarify whether or not macos-11 images are publicly available in this thread and in the readme?

Thanks!

**Comment by gereons** [comment]
@jpsim I struggled to understand this as well, but this snippet from https://github.com/actions/virtual-environments#available-environments explains it, afaict:

> The macOS 11 virtual environment is currently in preview, and is automatically available to existing Enterprise plan customers. New Enterprise plan customers, or customers on other plans, should fill the form to request access to macOS 11 virtual environment. Please view our Big Sur guide for more details.
The macos-latest YAML workflow label still uses the macOS 10.15 virtual environment.

I.e, macos-11 is currently only available for existing enterprise customers, and everyone else needs to apply for access using the form linked to in https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md

**Comment by OliverPearmain** [comment]
🍿

**Comment by jpsim** [comment]
Ah, so in reality [this comment](https://github.com/actions/virtual-environments/issues/2486#issuecomment-852906007) was just to share that there's nothing new to share 🙃

**Comment by JetForMe** [comment]
> * current Enterprise customers also have been granted access to macOS 11 pools.

What qualifies as "current?" We've been enterprise customers for over a year.

**Comment by ahdbilal** [comment]
Folks! if you don't have access to Big Sur, please fill this [form](https://forms.office.com/pages/responsepage.aspx?id=v4j5cvGGr0GRqy180BHbR_fA7dPeYFZNhXEED2yRSuVUQk01REoxS05JTUhZRDFMTkYzRjVURklWMy4u) and we will work quickly to give you the access. This is not a perfect solution but we decided to roll out Big Sur in phases instead of waiting for even more days and opening the pool to all. As we add more capacity, we will expand the preview to automatically include everyone on other paid and free plans. Thank you all for your patience.

**Comment by jeroenleenarts** [comment]
Submitted the form for our Azure DevOps org.

**Comment by Tovkal** [comment]
> Submitted the form for our Azure DevOps org.

@jeroenleenarts I think the preview is only for GitHub users. Us AzDO customers have to wait some more. See the comment bellow from Alena:

> macOS 11 is currently not available for Public AzDO customers, but we are also working on providing it as soon as possible. Azure DevOps infrastructure a bit different so the amount of work required is also different.

**Comment by jeroenleenarts** [comment]
We'll see, it is for an enterprise AzDO. So I expect some level of support beyond "hopefully some time soon".

**Comment by JetForMe** [comment]
> Folks! if you don't have access to Big Sur, please fill this [form](https://forms.office.com/pages/responsepage.aspx?id=v4j5cvGGr0GRqy180BHbR_fA7dPeYFZNhXEED2yRSuVUQk01REoxS05JTUhZRDFMTkYzRjVURklWMy4u) and we will work quickly to give you the access. This is not a perfect solution but we decided to roll out Big Sur in phases instead of waiting for even more days and opening the pool to all. As we add more capacity, we will expand the preview to automatically include everyone on other paid and free plans. Thank you all for your patience.

I filled out the form a couple days ago; have yet to hear anything so much as a confirmation email.

**Comment by vadi2** [comment]
Apple is notoriously developer-hostile and as far as I'm aware, there are no other CI/CD services giving such ease of access to a macOS system, nonetheless an M1 chip. Thus the work the Github team is doing here is rather ground-breaking - let's give them some time. Adding more pressure will not help them.

**Comment by tonyarnold** [comment]
> […] there are no other CI/CD services giving such ease of access to a macOS system […] 

Sorry, but this just isn't true. GitHub is significantly behind all of their competitors in this space. There are plenty of other CI/CD services that have been providing macOS 11 CI since late last year - here are three big ones that I found within about 60 seconds of searching:

- https://blog.bitrise.io/post/hosted-ci-xcode-support
- https://circleci.com/docs/2.0/testing-ios/#supported-xcode-versions
- https://buildkite.com/blog/introducing-buildkite-macos-ci-managed-service

> Thus the work the Github team is doing here is rather ground-breaking […] 

It's really not - see the links above. GitHub also haven't given any indication that they're deploying M1-based hardware for their hosted runners. 

I genuinely hope GitHub works through whatever challenges they are facing here, but based on the tiny, vague bit of information they've chose to share with us publicly, it feels an awful lot like they were caught very unprepared by ever having to upgrade the major macOS version of their CI. 

Anyway, it looks like they're slowly deploying macOS 11 now, so I guess we'll see in the coming weeks how they deal with macOS 12.

**Comment by AlenaSviridenko** [comment]
Hey, I have an update: everyone who have submitted the form before June 3rd, has been granted access to macOS 11 pool.

**Comment by gtalarico** [comment]
Thanks @AlenaSviridenko 🎉 

Just wanted to report back that I was able to use the new mac11 image.

Minor side note, looks like the image was renamed from `macos-11.0` to `macos-11` [here](https://github.com/actions/virtual-environments/commit/c60f54012ab18029c1efe8a74bca16ed89a25791#commitcomment-51693421).

But the json schema vscode uses still expecting `macos-11.0` so it complains. 
Not sure if the schema is updated automatically or not. I dropped a note here:
https://github.com/actions/virtual-environments/commit/c60f54012ab18029c1efe8a74bca16ed89a25791#commitcomment-51693421

**Comment by kneekey23** [comment]
I finally got the new mac11 image as well but it still has swift 5.3 on it?
```
> swift --version
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin20.5.0
```
Is anyone else experiencing this?

**Comment by jpsim** [comment]
The macOS 11 image has Xcode 12.4 selected by default. You’ll need to set it to Xcode 12.5 by setting `DEVELOPER_DIR` in your GitHub action workflow.

**Comment by kneekey23** [comment]
Thank you @jpsim !

**Comment by jeroenleenarts** [comment]
@jpsim Any indication on Azure Pipelines? I submitted the form as well, but that was on the 3rd and you indicated everyone submitted before that day receiving access to the latext macOS images. :)

**Comment by miketimofeev** [comment]
@kneekey23 @jpsim FYI: we have announced switching the default Xcode to 12.5 in 2 weeks https://github.com/actions/virtual-environments/issues/3522

**Comment by jeroenleenarts** [comment]
@miketimofeev How to tell if I have macOS 11 images available on Azure DevOps Pipelines?

**Comment by AlenaSviridenko** [comment]
Azure DevOps has a bit different timeline, progress can be tracked here https://github.com/actions/virtual-environments/issues/2072#issuecomment-847270559

**Comment by kemalenver** [comment]
> Hey, I have an update: everyone who have submitted the form before June 3rd, has been granted access to macOS 11 pool.

Any chance of checking the form again for people since June 3rd? :) My team were recently asked to ditch Circle and are now blocked because we can't access Xcode 12.5 via macos11.

**Comment by AlenaSviridenko** [comment]
Hi,
yep, we just granted access to those who submitted the form by the end of June, 6. 
I am monitoring capacity to define when we can do the new round of pool opening.

**Comment by sisoje** [comment]
> Hi,
> yep, we just granted access to those who submitted the form by the end of June, 6.
> I am monitoring capacity to define when we can do the new round of pool opening.

we did not get the access, even though I submitted before June 6th

**Comment by harlantwood** [comment]
We also submitted the form before June 6, and don't have access to `macos-11` on our (private) repo: 

https://github.com/wikinodes-net/wikinodes

The message we got 3 hours ago: `No runner matching the specified labels was found: macos-11`

Thanks for your efforts, much appreciated!

**Comment by AlenaSviridenko** [comment]
Hi @sisoje, I see you have submitted it several times. We will investigate why the image is still not available for you.

**Comment by JetForMe** [comment]
Thank you so much @AlenaSviridenko. I was able to run a small script on macOS 11 and pass our tests. This is going to be great. Thanks!

**Comment by lennartstolz** [comment]
@AlenaSviridenko Sorry for bothering you.
I just submitted the form as well and I'd love to get access granted - I'd love to generate `DocC` documentation with Xcode 13 using Github actions.

**Comment by jessesquires** [comment]
> we did not get the access, even though I submitted before June 6th

This is also the case for me, unfortunately. cc @AlenaSviridenko I have tried re-submitting.

**Comment by miketimofeev** [comment]
@erikng the image with Xcode 13.0 will be deployed today

**Comment by simondelphia** [comment]
Does `macos-latest` point to macOS 11 if you're on the preview?

**Comment by miketimofeev** [comment]
@simondelphia no, we don't have a plan to switch the label at the nearest future

**Comment by einar-notland** [comment]
We also need access to macOS 11.0 pool. Seems our apps cannot run on iOS 15 beta because when built with xcode 12.4, so we need xcode 12.5 for our build-jobs, at least before iOS 15 goes into public beta.

Submitted the form today, because we were not aware of "the form" earlier.

**Comment by akago-dev** [comment]
Where is this form ?

**Comment by radarhere** [comment]
@akago-dev
> I.e, `macos-11` is currently only available for existing enterprise customers, and everyone else needs to apply for access using the form linked to in https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md

**Comment by vcunat** [comment]
A dumb question: are the macos-11 runners x86_64 only?  (I'm sorry, I've spent lots of time searching docs and trying out without finding explicit information.)

**Comment by hakonk** [comment]
According to this [document](https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md), "Big Sur will automatically become available to existing Enterprise customers". Does this mean _all_ existing enterprise customers will get access automatically without submitting the form, or does it mean existing customers will get access granted they submit the said form? If this becomes available to all customers automatically, when will this happen?

**Comment by AlenaSviridenko** [comment]
Hi @hakonk,
we update a note in the main README file with some clarification what does it mean:
`The macOS 11 virtual environment is currently in preview and is automatically available to the existing Enterprise plan customers who used macOS hosted runners at least once between May, 1 - June, 1. New Enterprise plan customers, or customers on other plans, should fill the form to request access to macOS 11 virtual environment. `

**Comment by ansp** [comment]
@AlenaSviridenko I submitted a form, how long approval process is taking?

**Comment by hakonk** [comment]
Thank you for the clarification, @AlenaSviridenko. 

With the following

```yml
pool:
  vmImage: macos-11
```

I get `##[warning]An image label with the label macos-11 does not exist.`.

Am I referencing the wrong image, or does this mean the organization has not (yet) gotten the access to the macos-11 pool?

**Comment by AaronMT** [comment]
Hello, 

Is there any estimate as to when `macos-latest` will transition to macOS 11?

**Comment by IvanTsanev89** [comment]
Hi guys,

We have 3rd party dependencies running on the latest Xcode that requires macOS 11. 
Our CI process is blocked by this and we need the OS released as soon as possible. We have filled out the form but still no response.
The OS was released 7 months ago from now.

Can we please get this version out so we can resume work as normal?
Thanks

**Comment by AlenaSviridenko** [comment]
Hi!
Enterprises who submitted the form before June, 15, have been granted the access to the pool. @IvanTsanev89 your organization was in this list.

@hakonk image label is correct, looks like org doesn't have access to the pool. Have you submitted the form or did your organization use GitHub-hosted macOS runners during the specified period?

@ansp we are reviewing the responses at least twice a week, but onboarding depends on current capacity situation.

@AaronMT not earlier than it goes out of preview. I'd say approximately transition will take place at the end of summer - mid of the fall, we are working on the plan.

**Comment by dfalling** [comment]
I'm still seeing the `No runner matching the specified labels was found: macos-11.0` failure.

All I've changed in my workflow is:
```yaml
jobs:
  test:
    runs-on: macos-11.0
```

Do I also need to add `pool`, `vmImage` like hakonk posted above? I can't find any documentation about those for Github Actions (just Azure), and my workflow fails when I add them to my job.

**Comment by beeradmoore** [comment]
@AlenaSviridenko , is there any ETA on when this should be moving out of preview?

**Comment by hakonk** [comment]
@AlenaSviridenko We submitted the form yesterday around 10:30.

**Comment by trr-amsiq** [comment]
> ```yaml
>  macos-11.0
> ```

should be macos-11 not macos-11.0

**Comment by dfalling** [comment]
It's failing for me with `macos-11` as well. I submitted the form last week so is there anything else I need to do to enable the pool?

**Comment by KarlHarnois** [comment]
I'm getting a similar error:
```
No runner matching the specified labels was found: macos-11
```

**Comment by leiless** [comment]
Hey, guys, I managed to use macOS 11.0 build toolchain in macOS 10.x for cross-compiling(build `arm64` binary under `x86_64` machine)

```bash
# ...

# Hack macOS 10.x to allow arm64* build support
if [ "$(sw_vers -productVersion | cut -d'.' -f1)" = 10 ]; then
  pushd /Library/Developer/CommandLineTools/SDKs
  sudo rm MacOSX.sdk
  sudo ln -s MacOSX11.1.sdk MacOSX.sdk
  sudo rm -rf MacOSX10.15.sdk
  ls -l
  popd
fi

# ...
```

Source: https://github.com/leiless/trojan-darwin-universal/blob/main/.github/workflows/build.yml#L40-L48

A bit of note, this hack is transient and provides no warranty or support. use at your own risk.

**Comment by AlenaSviridenko** [comment]
Hey,
Update: access is granted for those who submitted the form before June, 17. Those, whom I emailed with the clarification, will be included to the next iteration.

Thanks.

**Comment by erikng** [comment]
FYI for others but email was marked as spam on my end.

**Comment by JoshCheek** [comment]
Appreciate the work y'all are putting into this, @AlenaSviridenko.

If it doesn't require a ton of effort, I think it'd be a lot smoother for people trying out GH Actions if the error message told us we didn't have access this environment. As is, I read the error message and then spent a lot of time analyzing my config trying to figure out what I did wrong:

<img width="540" alt="image" src="https://user-images.githubusercontent.com/77495/122687290-432d0900-d1db-11eb-84ca-58171154cf60.png">

But, I also know that may not be a trivial thing to do, so if not 🤷 such is life.

**Comment by tsapeta** [comment]
@JoshCheek, the correct label for macOS 11 runner is `macos-11`, not `macos-11.0` — maybe that's why it's failing for you. See https://github.com/actions/virtual-environments#available-environments

**Comment by JoshCheek** [comment]
> @JoshCheek, the correct label for macOS 11 runner is macos-11, not macos-11.0 — maybe that's why it's failing for you. See > https://github.com/actions/virtual-environments#available-environments
> That was also mentioned a few times in this thread 👀

Thanks :) I did wind up trying both of those. I tried a lot of things for several hours and in the end, I returned to `macos-11.0`, because that's what the step I used to set up Ruby used ([here](https://github.com/ruby/setup-ruby/blob/477b21f02be01bcb8030d50f37cfec92bfa615b6/.github/workflows/test.yml#L18)), which seems to be working for them (eg [here](https://github.com/ruby/setup-ruby/runs/2867896686?check_suite_focus=true)). Appreciate the suggestion, though.

---

I guess this is an example of what I mean about better error messages. We can't tell from this error whether I've configured it wrong or whether I don't have access to this environment. So it's unclear whether to continue troubleshooting or not.

---

🤔 You know! 😃 I may have thought of an easy way to handle this: **appending the list of available runners to the error message!**

Presumably GH Actions checked for this "runner" from a set of runners. So if it doesn't match any of them, it could say "...and here are the runners / labels that I _do_ know about". Then it would be trivial to determine whether you have it misconfigured (and what it should be configured to), or whether you lack permission.

**Comment by marcbaldwin** [comment]
Hi @AlenaSviridenko, I filled in the form on behalf of my organisation but have yet to receive a response. Please could you grant us access to the macos-11 runner? My email is marc@beeline.co 🙏

**Comment by AlenaSviridenko** [comment]
Hi @marcbaldwin, you org should have been received access on Thursday, please, check it. Unfortunately, we don't yet have any individual informing way when access is granted, so I am posting updates here in thread when including another set of accounts.

**Comment by AlenaSviridenko** [comment]
👋 hi!
Update: macOS 11 pool is enabled for those who submitted the form before June, 22.

Thanks!

**Comment by marcbaldwin** [comment]
> Hi @marcbaldwin, you org should have been received access on Thursday, please, check it. Unfortunately, we don't yet have any individual informing way when access is granted, so I am posting updates here in thread when including another set of accounts.

Thanks so much @AlenaSviridenko 🙏

**Comment by radarhere** [comment]
@AlenaSviridenko hi. Thanks for managing this. My turn to ask - I've submitted the form three times on different dates for the python-pillow account, but it's not working yet and I've received no response. Is there any information you can give me?

**Comment by AlenaSviridenko** [comment]
@radarhere what is the image label you are using? Correct image label is `macos-11`

**Comment by radarhere** [comment]
@AlenaSviridenko you're right. Thanks. It works now. Apologies for missing that.

**Comment by jimmyhoran** [comment]
@AlenaSviridenko firstly thanks for managing this push to macOS-11. I have a question regarding Xcode 12.5.1 for `macos-11`. I can see that support for Xcode 12.5.1 has already been added to `images/macos/toolsets/toolset-11.json`. I was wondering when we might see the `macos-11` label pointing to the recent `toolset-11.json` changes. So that those of us with access to `macos-11` can explicitly make use of Xcode 12.5.1. Thanks in advance 🙏

**Comment by AlenaSviridenko** [comment]
Hi!
Update: access has been granted for those who submitted the form before June, 24th.

@jimmyhoran, answering your question, new image will be deployed next week, likely on Tuesday. Usually, it takes 4-5 business days from new Xcode release to having it on our images.

**Comment by Specineff00** [comment]
Hi @AlenaSviridenko where would I need to go to fill out this form?

**Comment by AlenaSviridenko** [comment]
Hi @Specineff00,
here is the link to the guide https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md, I have also updated issue description for visibility.

**Comment by jahmai-ca** [comment]
> Hi!
> Update: access has been granted for those who submitted the form before June, 24th.
> 
> @jimmyhoran, answering your question, new image will be deployed next week, likely on Tuesday. Usually, it takes 4-5 business days from new Xcode release to having it on our images.

I requested access using the form before June 24th and I don't have access.

**Comment by AlenaSviridenko** [comment]
@jahmai-ca, that is strange, all logs show that you should have access. What is the image label you are using? The correct one is `macos-11`

**Comment by jahmai-ca** [comment]
@AlenaSviridenko 

![image](https://user-images.githubusercontent.com/2010997/123641127-407a8580-d865-11eb-8034-76f24972e283.png)

Is it because it's been applied to my personal github account and not my companies? (criticalarc)

**Comment by AlenaSviridenko** [comment]
@jahmai-ca, oh, you are using Azure DevOps. I assumed it is https://github.com/criticalarc and enabled it in GitHub Actions for this org. For Azure Pipelines timeline and process is a bit different, please see this comment for details https://github.com/actions/virtual-environments/issues/2072#issuecomment-854812898.

**Comment by juliand665** [comment]
I've been following this thread for the longest time so I'll get notified once macOS 11 is fully supported, but the constant emails are kinda getting to me. Is there anywhere I can sign up to be notified when macOS 11 support finally reaches everyone?

**Comment by jsumners** [comment]
> I've been following this thread for the longest time so I'll get notified once macOS 11 is fully supported, but the constant emails are kinda getting to me. Is there anywhere I can sign up to be notified when macOS 11 support finally reaches everyone?

I'm kind of with you. It's been frustrating getting a notification on this issue for everyone who doesn't read the issue and posts the same thing again. I just want to be aware of when I can start using this in a small OSS project.

**Comment by AlenaSviridenko** [comment]
@juliand665, I can suggest customizing notifications to be notified only when it is closed, because I will definitely close that issue when macOS 11 will become generally available 🙂
![image](https://user-images.githubusercontent.com/8777209/123654846-23c75900-d837-11eb-8edf-2fcb9967a161.png)

**Comment by JustinYeoh93** [comment]
Hi there,

Thanks everyone in the team for getting this up as soon as possible.
I went through the on-boarding guide and signed up.

I was wondering how would I know if I am approved for access?
Will there be like a mail notification or something related?

Regards,
Justin

**Comment by AlenaSviridenko** [comment]
Hi!
macOS 11 pool is enabled for those who submitted the form before June, 29th. I have sent several emails with clarification questions, these accounts will be included to the next iteration.
Thanks!

**Comment by acenario** [comment]
Hi, I submitted the form awhile ago and didn't hear back. Were we added?

**Comment by jobertcodes** [comment]
@AlenaSviridenko First of all, I appreciate the efforts to support macOS 11.  Would you have a timeline when to expect this to be generally available?

**Comment by AlenaSviridenko** [comment]
@acenario, what is the organization name you have submitted for?

**Comment by acenario** [comment]
> @acenario, what is the organization name you have submitted for?

"Ira Jaan"

**Comment by AlenaSviridenko** [comment]
@acenario, yes, 2 weeks ago it was enabled it for your organization. Unfortunately, there is no personal informing way yet, so I am posting updates here.
Just a reminder, correct image label is `macos-11`

**Comment by AlenaSviridenko** [comment]
Hey,
update: macOS 11 is enabled for those who have submitted the form before July, 1st. I have also sent several clarification emails, these accounts will be included to the next iteration. 
We also have a code freeze next week, so the next iteration will take place a week of July, 14th.

Thanks.

**Comment by boneskull** [comment]
@AlenaSviridenko thank you!

**Comment by lkuczborski** [comment]
Any updates on this? When we will be able to use this environment? It is impossible to use Xcode 13 Beta or even Xcode 12.5/12.5.1 stable without it :(

**Comment by kneekey23** [comment]
+1 on Xcode 13 beta. Is there a timeline for that?

**Comment by AlenaSviridenko** [comment]
Hi, Xcode 13 beta is available on macOS 11 image https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode. Have you submitted the form to be able to use that image?

**Comment by kneekey23** [comment]
I have access to the Mac OS 11 image already. Is there another form for Xcode beta 13 separately?

**Comment by rinsuki** [comment]
@kneekey23 you can use Xcode 13 beta in macOS 11 image. just `run: sudo xcode-select --switch /Applications/Xcode_13.0.app`

**Comment by lkuczborski** [comment]
@AlenaSviridenko is the macos11 preview available for Individuals too or only Organizations?

**Comment by AlenaSviridenko** [comment]
@lkuczborski for both organizations and individuals. Individual account name and organization name are treated the same way, so you can fulfill the form, and macOS 11 will be enabled for you once we have capacity for it.

**Comment by kneekey23** [comment]
Thanks! Will do!

**Comment by lkuczborski** [comment]
@AlenaSviridenko thanks, submitted!

**Comment by davdroman** [comment]
@AlenaSviridenko I submitted my name (@davdroman) and my company's name (@rythmico) only a day or two after the form came out last month and I still don't have it enabled. Would you be able to verify please?

**Comment by AlenaSviridenko** [comment]
@davdroman for organization, macOS 11 should work with label `macos-11`, but for your personal account I cannot find any confirmation that it is enabled. I will make sure to include you to the new iteration next week. I apologize for inconveniences, perhaps overlooked that as we have a lot of requests.

**Comment by xavier2k6** [comment]
@AlenaSviridenko I've made a submission today for our FOSS organization `qbittorrent`, so hopefully it can be included in the new iteration next week.

**Comment by AlenaSviridenko** [comment]
Hi,
update after a silent week: macOS 11 was enabled for those who submitted the form before July, 12th. I have also sent several clarification emails. These organizations/accounts will be included to the next iteration once capacity is ready.

Thanks.

**Comment by cquanu** [comment]
Hi @AlenaSviridenko, is this for GitHub Actions or Microsoft App Center?

**Comment by arbourd** [comment]
This might be too a big an ask but, perhaps in the future for "Private previews" we could see an error message in the runner itself. I was about to open a ticket because of this:

<img width="521" alt="Screen Shot 2021-07-13 at 12 37 13 PM" src="https://user-images.githubusercontent.com/7211830/125491515-288bedb8-9331-43f4-a86e-752e21e84f23.png">

As it works fine on personal macos-11 projects but not my company's enterprise account.

**Comment by xavier2k6** [comment]
> @AlenaSviridenko I've made a submission today for our FOSS organization @qbittorrent, so hopefully it can be included in the new iteration next week.

@AlenaSviridenko Are we added to the list??
Also, is it just for the qbittorrent/qBittorrent organization or do the team members like myself have it activated for our own forked repositories?

**Comment by AlenaSviridenko** [comment]
@cquanu , it is for GitHub Actions.

@xavier2k6 , yes, qbittorrent org was also included to preview. Only the original org, the form should be submitted for each account that forked the repo.

**Comment by xavier2k6** [comment]
@AlenaSviridenko Thank you for the clarification, have advised team-members to request access if needed via the form.

**Comment by bdolewski** [comment]
@AlenaSviridenko 

> We will review the incoming requests periodically and open macOS 11 pool for some amount of them

How long does it take your team to process requests? My company just switched from GitLab to Azure in July. We don't have access to `macos-11` image but we need it urgently since it contains Xcode 12.5 with simulators for iOS 14.5. 

Our main iOS project has deployment target set for iOS 14.5 and we cannot downgrade it to iOS 14.4 (which is the top version supported in `macos-latest`).

**Comment by AlenaSviridenko** [comment]
@bdolewski for GitHub Actions, I am usually reviewing requests 2-3 times a week, but enabling macOS 11 two times a week. For Azure DevOps, process is a bit different: https://github.com/actions/virtual-environments/issues/2072#issuecomment-854812898. So far, macOS 11 is enabled for Azure DevOps organization once a week.

**Comment by AlenaSviridenko** [comment]
Another update: macOS 11 access has been granted for those who submitted the form before July, 15th.

**Comment by Brett-Best** [comment]
@AlenaSviridenko
https://github.com/Brett-Best/SwiftUIX/actions/runs/1032626590
> No runner matching the specified labels was found: macos-11.0

https://github.com/SwiftUIX/SwiftUIX/runs/3071167967
> Works fine

I’ve noticed that in my forked version of SwiftUIX if I use the exact same label as the original repo, I get the no runner error.
However, if I change the runner to `macos-11`, it works in my fork.

Why does `macos-11.0` work in the original repo `SwiftUIX/SwiftUIX` but in my fork `Brett-Best/SwiftUIX` it does not?

**Comment by AlenaSviridenko** [comment]
@Brett-Best when we started opening macOS 11 pool by the form requests, we also renamed runner to macos-11, because it is more correct given the new macOS versions naming. 
Your organization probably used macOS 11 runner before that update, and we added backward compatibility for such cases.
I would suggest changing runner label in the original repo for consistency.

**Comment by AlenaSviridenko** [comment]
Hi there,
Those who submitted the form before July, 19th have been granted the access to macOS 11 pool.

**Comment by arturonaredo** [comment]
Hi @AlenaSviridenko,

is there any timeline when this will available to Azure Devops? I already tried to use it for my Enterprise with no success. Also reached my representative. 

Thanks.

**Comment by AlenaSviridenko** [comment]
Hey!
Those who submitted the form before July, 21st have been granted the access.
I have sent several clarifying emails, they may land in a spam folder, please, check. 

Thanks.

**Comment by arturonaredo** [comment]
Hi @AlenaSviridenko ,

Yep, I just submited the form. My question is about using this images in Azure Devops not github actions. There is no updated documentation in the hosted reference page: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml so I'm not clear on what the YAML VM Image Label will be neither this will work to Azure Devops environment.

Thanks in advance.

**Comment by AlenaSviridenko** [comment]
Hi @arturonaredo,
the form is only about GitHub Actions. Azure DevOps process is tracked in another issue and onboarding process is described here https://github.com/actions/virtual-environments/issues/2072#issuecomment-854812898. New Azure DevOps orgs are onboarded about 2-3 times a month.
Unfortunately, we still have more serious capacity issues for Azure DevOps pools, so macOS 11 will remain in such preview state a bit longer than for GitHub Actions. 
Documentation will be updated after opening the pool for everyone.

**Comment by backslash-f** [comment]
Hi @AlenaSviridenko, we submitted the form before July, 21st and we still don't have access to it. Nothing in my spam folder, could you please check? Our enterprise account is https://github.com/labtwin Thank you.

**Comment by AlenaSviridenko** [comment]
Hey,
new update: those, who submitted the form before July, 22nd, got access to macOS 11.
Thanks.

**Comment by AlenaSviridenko** [comment]
👋 hi!
We are planning to open Big Sur pool for all by the end of next week, but this is for GitHub Actions only. We have escalated requests about Azure DevOps Big Sur, because it is also critical. Hopefully, it will be open soon, too. 
Thanks.

**Comment by GabrielHauber** [comment]
Submitted form early June I think it was (had a response back on 8 June asking for clarification on org name) and I don't yet seem to have access…

**Comment by Tokenyet** [comment]
This should be added in [docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) since this issue keeps going, should make a notice in docs to prevent confusion.

**Comment by AlenaSviridenko** [comment]
Hi,
access has been granted for those, who submitted the form before July, 27th.
Thanks.

**Comment by anfriis** [comment]
@AlenaSviridenko 
Can you give an update and ETA on App Center support?
We urgently need Xcode 12.5 support in App Center to be able to build our app, else we will have to move our organization to another build service.
No one seems to be able to give an update in the App Center issue related to this: https://github.com/microsoft/appcenter/issues/2185

**Comment by zeryen** [comment]
Hello! It's critical for everyone and App Center doesn't give a thing to this. The issues is already open since 4 may and no updates, no ETA. I hope you are loosing customers with such attitude...

**Comment by AlenaSviridenko** [comment]
Hi 👋
as for the App Center — it shares machines with Azure DevOps, please, see the latest update about Azure DevOps offering Big Sur https://github.com/actions/virtual-environments/issues/2072#issuecomment-888037580

**Comment by mi-mazouz** [comment]
How can I get access to masOS-11 image plase?

**Comment by AlenaSviridenko** [comment]
Hi @mi-mazouz, please check out this guide on how to get access: https://github.com/actions/virtual-environments/blob/main/docs/macos-11-onboarding.md

**Comment by AlenaSviridenko** [comment]
Update: access has been granted for those who submitted the form before July, 28th.

**Comment by mi-mazouz** [comment]
Thank you @AlenaSviridenko just submitted the form!

**Comment by mi-mazouz** [comment]
@AlenaSviridenko Sorry but any update, Still fail on macos-11 image

**Comment by AlenaSviridenko** [comment]
Hi @mi-mazouz, next iteration will occur on Monday, I'll post an update here. Unfortunately, was not able to enable macOS 11 pool yesterday due to an incident.

**Comment by AlenaSviridenko** [comment]
Hi there,
those, who submitted the form before August, 2, has been granted access to macOS 11.

**Comment by thotasairam** [comment]
@AlenaSviridenko I have just submitted the request. One quick question, once I get access to macOS 11, will this allow me to run 12.5 builds on App Center? Or do I have to use git hub actions for the CI/CD of my react native app?

**Comment by AlenaSviridenko** [comment]
@thotasairam, this is only for GitHub Actions, App Center doesn't support Big Sur now.

**Comment by AlenaSviridenko** [comment]
👋
another round of approvals: access has been granted to those who submitted the form before August, 3.
Thanks.

**Comment by scannillo** [comment]
👋  Hello. We just filled out the form today (PayPal org). We're curious how frequently you're doing rounds of approvals so we can plan for when this will be available to use. Thank you!

**Comment by AlenaSviridenko** [comment]
👋 hi,
Access has been granted for those who submitted the form before August, 4. I have sent several clarification email regarding some requests, please, check them.
Thanks!

**Comment by AlenaSviridenko** [comment]
👋
new train of approvals just arrived 🚉, access was granted for those who submitted the form before August, 5.
Thanks.

**Comment by MaxDesiatov** [comment]
The form currently requires you to specify a GitHub organization. Does this mean that macOS 11 is not available for individual repositories? When could we expect it to become available then?

**Comment by AlenaSviridenko** [comment]
@MaxDesiatov, personal account name is also treated like organization name. We are enabling it on the organization/account level, not a certain repository. So feel free to submit the form, so we can include you in the preview pool.
Thanks.

**Comment by brysonreece** [comment]
Is there any update on the general availability of this? I see that there were plans to release during this last week, but I'm assuming it got pushed based on the need to still request access.

**Comment by miketimofeev** [comment]
@brysonreece unfortunately, we're a bit delayed for now, probably 1 or 2 weeks.

**Comment by MaxDesiatov** [comment]
Since macOS 12.0 is only a month or two away, should we expect the same very delayed rollout for that version, with it becoming generally available only in August of 2022?

**Comment by miketimofeev** [comment]
@MaxDesiatov we're working on delivering macos-12 on time (September-October)

**Comment by miketimofeev** [comment]
The access was granted to everyone who submitted the form before August, 9 — 9.00 UTC

**Comment by michaelpeternell** [comment]
Why not just approve it for everyone?

**Comment by michaelpeternell** [comment]
I'm really disappointed of you guys, I will start evaluating github alternatives..

**Comment by badhenke** [comment]
@AlenaSviridenko Hey! We (molistreaming) just made a request through the form. Hope to get access in next approval-train. Thanks for your hard work!

**Comment by miketimofeev** [comment]
@badhenke the access is granted to your org and to all the others who submitted the form before August 10, 15-00 UTC

 I have also sent several emails regarding some requests, please, check them.

**Comment by noisypigeon** [comment]
@miketimofeev Hey there! Just missed this train, when will the next one be happening? :) We (@valora-inc) would love to have access for our builds please. Thank you!

**Comment by miketimofeev** [comment]
Hey @noisypigeon! It will be in the next 24 hours.

**Comment by tonyarnold** [comment]
> we're working on delivering macos-12 on time (September-October)

I'd like to point out that making macOS 12 images available after the OS is released is _not_ "on time". 

We need to be able to test and run our app CI using the beta releases of these OSes, and using Xcode 13.

**Comment by kode54** [comment]
Here's the thing. macOS 11 made a huge hurdle with both its new licensing policies and with the option of rolling out ARM hardware. macOS 11 took all this time to roll out in a meaningful capacity, and other businesses who rolled it out faster had to comply with newer policies, such as requiring individuals or organizations to pay for a minimum of 24 hours on the same hardware at a time.

The transition to macOS 12 will not require any meaningful changes over the transition to 11. Therefore, as soon as the transition to 11 is complete, it will be on the roadmap to transition to 12 when it actually releases to stable, which is in September or October, depending on Apple.

**Comment by miketimofeev** [comment]
Access is granted for those who submitted the form before August 12, 09-00 UTC

**Comment by noisypigeon** [comment]
Hey @miketimofeev Thanks for giving access to @valora-inc. I've requested access for a personal project I'm working on with friends. I'm on a GitHub Teams plan so hopefully I'm eligible? Thanks for all the work y'all do!

**Comment by miketimofeev** [comment]
@noisypigeon hey! Sent you an e-mail with a clarification question.

**Comment by noisypigeon** [comment]
@miketimofeev Thanks! Replied!

**Comment by miketimofeev** [comment]
Access is granted for those who submitted the form before August 13, 12-00 UTC

**Comment by noisypigeon** [comment]
@tonyarnold Sharing this as an FYI. If you need macOS 12 sooner, you can get a Mac Mini from Mac Stadium (for free if your project is open source) and request that they install macOS 12 on it and then use the self runner. Obviously not as great as GitHub Actions but I'm looking into this for my own project and thought I'd share. :) 

@kode54 thank you for saying this! as someone who's previously worked in infra (i'm not a gh employee) there's so much that goes into offering the apple ecosystem to devs. I wish apple would make it easier on everyone but oh well hahahah. Thanks for everything y'all do!

**Comment by tonyarnold** [comment]
Thanks for the info, @noisypigeon. I transitioned to using a couple of Mac minis in our office with self-hosted runners back when this preview started. The performance issues were too much for my team to keep working, so we worked around it. 

Good to know that MacStadium is doing free for OSS!

**Comment by CamJN** [comment]
There doesn't seem to be a free tier... https://www.macstadium.com/pricing

**Comment by noisypigeon** [comment]
@tonyarnold yeahhhh, infra is not fun to maintain haha. Big part of my current role so I tolerate it in my personal projects. 🙈 

@CamJN You have to apply but from what I heard most people get approved: https://www.macstadium.com/opensource.

**Comment by dandre-hound** [comment]
@miketimofeev any idea when the next form submission review will? @tradehounds would like access to macos-11 for a build we're looking to push early this week and we just missed the Friday access review.

**Comment by miketimofeev** [comment]
We are happy to announce that macOS-11 is out of preview and now generally available for all the GitHub customers 🥳 
The docs will be updated later this week.

**Comment by cquanu** [comment]
> We are happy to announce that macOS-11 is out of preview and now generally available for all the GitHub customers 🥳 
> 
> The docs will be updated later this week.

@miketimofeev does it mean is for AppCenter too? Are we getting XCode 12.5?

**Comment by miketimofeev** [comment]
@cquanu unfortunately, no. This is still in progress and can be tracked here https://github.com/actions/virtual-environments/issues/2072

**Comment by Cheesebaron** [comment]
Azure Pipelines support is when?

![image](https://user-images.githubusercontent.com/249719/129770327-69c08f17-ad9e-4ef4-ac1e-2b3a67db5902.png)

Hosted agent docs for pipelines lead here.

**Comment by kode54** [comment]
Maybe case sensitive? I believe the OS tags are supposed to be all lower case?

**Comment by miketimofeev** [comment]
@Cheesebaron all the discussion about Azure Pipelines are here https://github.com/actions/virtual-environments/issues/2072

**Comment by miketimofeev** [comment]
I'm going to close this issue as the blog post is published https://github.blog/changelog/2021-08-16-github-actions-macos-11-big-sur-is-generally-available-on-github-hosted-runners/ and the Readme on the main page is updated as well:
![image](https://user-images.githubusercontent.com/48208649/129808020-695368f5-d53e-4e9d-a682-58d8cc9af75c.png)

**Comment by abitmore** [comment]
FWIW we need to update our workflows from
```
runs-on: macos-11.0
```
to
```
runs-on: macos-11
```

**Comment by miketimofeev** [comment]
@abitmore sorry for the inconvenience, we should have announced it.

**Comment by andreineculau** [comment]
@miketimofeev maybe this should go on a separate thread, but can you clarify why you took this decision to go `macos-11.0` to `macos-11`, given that the current version is actually 11.5? Historically what was more important/granular was actually the Xcode version, and only secondarily the macos version  - e.g. https://docs.travis-ci.com/user/reference/osx/#macos-version . I have been caught red-handed by this switch earlier today (which means it's now impossible to re-run a workflow that references `macos-11.0`, which in my books it's a big no-no), and after reading the blogpost I thought "This is definitely a glitch", but your reply above is a confirmation that this is not a glitch but intent. Thanks

**Comment by maxim-lobanov** [comment]
@andreineculau , We provide the single macOS 11 image (`macos-11`) that has latest 11.x updates (currently, it is 11.5). When we say `macos-11`, it means latest `11.x`.

Initially, `macos-11.0` label was created by mistake under impression that Apple follow the same approach like 10.13, 10.14, 10.15. With the release of macOS 11.1 we have fixed it and created `macos-11` label to not confuse users (because `macos-11.0` actually contained macOS 11.1 that doesn't sound correct, right?). `macos-11.0` was kept for backward compatibility.

> I have been caught red-handed by this switch earlier today (which means it's now impossible to re-run a workflow that references macos-11.0, which in my books it's a big no-no)

Sorry for inconvenience, `macos-11.0` was disabled by mistake yesterday during opening macOS 11 pools and this issue is already fixed for now. Currently, the label should work and we will keep it for backward compatibility.

As for the Xcode version vs macOS version, we don't provide the separate image for every Xcode so it won't work for us.
The list of Xcode versions installed on macOS 11 can be found in documentation: https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode

**Comment by andreineculau** [comment]
@maxim-lobanov Thanks for the clear explanation and confirmation that you'll keep macos-11.0 for backward compatibility!
